Sage/carbon
# Adds icons to form summary (abstraction too)

## Description

* this sub-component sits in the Form folder
* will handle the output of errors and warnings, with icons
* is also applied to the Form component with some complexity removed

* Cancel button test is changed to use class rather than array index, there's no need to check `length` with `TestUtils.find...`
* Tests for form summary contents are removed, we now check that the component is renedred with the correct props
* Enzyme is also added in to cover changed tests for FormSummary within Form

* `&nbsp;` is prefered to margin for text spacing for flexibility
* icon alignment is not absolutely perfect, it looks like the warning icon might need a tweak

## TODO

- [x] Design review (we just did this as designs weren't available)

# Screenshots / Gifs
![image](https://cloud.githubusercontent.com/assets/10499070/24050495/e2abd162-0b26-11e7-8e85-b1c441dc4b94.png)
![image](https://cloud.githubusercontent.com/assets/10499070/24050543/09b22d74-0b27-11e7-8826-542b91c5b548.png)

![image](https://cloud.githubusercontent.com/assets/10499070/24050437/b0f50cf6-0b26-11e7-94e2-3344a985039b.png)


## QA notes

**bear in mind form state might not match summary on the QA branches, we're only interested in the design of the summary**

This work has been branched off release which means the old demo site is displayed, once gulp is run you'll find that at [localhost:8080](http://localhost:8080/forms)

You can test errors by clicking Save on the form example at the top of the page. For warnings only use [this QA branch](https://github.com/Sage/carbon/tree/abstract-out-form-summary-message--warnings-only-QA-branch).

To test a summary message that contains warnings use [this QA branch](https://github.com/Sage/carbon/tree/abstract-out-form-summary-message--errors-and-warnings-hack-for-qa) - switch back to the other branch to test errors only or a clean form validation pass

On successful QA pass please delete both the QA branches [one](https://github.com/Sage/carbon/tree/abstract-out-form-summary-message--warnings-only-QA-branch) and [two](https://github.com/Sage/carbon/tree/abstract-out-form-summary-message--errors-and-warnings-hack-for-qa)
